### PR TITLE
Update feast transformer docs

### LINKED ISSUE #2729: Feast transform example not working
/kind bug

**What steps did you take and what happened:**
I'm trying to run the feast transform example from docs... but I got some errors in the way.

Container sklearn-driver-transformer-transformer-default-00001-    log

kserve-container [I 230303 08:30:40 driver_transformer:61] Model name = sklearn-driver-transformer                                                                                                         
kserve-container [I 230303 08:30:40 driver_transformer:62] Protocol = v1                                                                                                                                   
kserve-container [I 230303 08:30:40 driver_transformer:63] Predictor host = sklearn-driver-transformer-predictor-default.default                                                                           
kserve-container [I 230303 08:30:40 driver_transformer:64] Feast serving URL = 172.18.255.200:6566                                                                                                         
kserve-container [I 230303 08:30:40 driver_transformer:65] Entity ids = ['driver_id']                                                                                                                      
kserve-container [I 230303 08:30:40 driver_transformer:66] Feature refs = ['driver_hourly_stats:acc_rate', 'driver_hourly_stats:avg_daily_trips', 'driver_hourly_stats:conv_rate']                         
kserve-container [I 230303 08:30:40 model_server:150] Registering model: sklearn-driver-transformer                                                                                                        
kserve-container [I 230303 08:30:40 model_server:123] Listening on port 8080                                                                                                                               
kserve-container [I 230303 08:30:40 model_server:125] Will fork 1 workers                                                                                                                                  
kserve-container [I 230303 08:30:40 model_server:128] Setting max asyncio worker threads as 20                                                                                                             
kserve-container [I 230303 08:31:33 driver_transformer:149] The online feature rest request status is 200                                                                                            
kserve-container [I 230303 08:31:33 driver_transformer:153] The input for model predict is {'instances': [[None, None, None, 1001], [None, None, None, 1002], [None, None, None, 1003], [None, None, None, 
kserve-container [E 230303 08:31:33 web:1798] Uncaught exception POST /v1/models/sklearn-driver-transformer:predict (127.0.0.1)                                                                            
kserve-container     HTTPServerRequest(protocol='http', host='sklearn-driver-transformer-transformer-default.default.svc.cluster.local', method='POST', uri='/v1/models/sklearn-driver-transformer:predict 
kserve-container     Traceback (most recent call last):                                                                                                                                                    
kserve-container       File "/usr/local/lib/python3.8/site-packages/tornado/web.py", line 1713, in _execute                                                                                                
kserve-container         result = await result                                                                                                                                                             
kserve-container       File "/usr/local/lib/python3.8/site-packages/kserve/handlers/predict.py", line 70, in post                                                                                          
kserve-container         response = await model(body)                                                                                                                                                      
kserve-container       File "/usr/local/lib/python3.8/site-packages/kserve/model.py", line 86, in __call__                                                                                                 
kserve-container         response = (await self.predict(request)) if inspect.iscoroutinefunction(self.predict) \                                                                                           
kserve-container       File "/usr/local/lib/python3.8/site-packages/kserve/model.py", line 211, in predict                                                                                                 
kserve-container         return await self._http_predict(request)                                                                                                                                          
kserve-container       File "/usr/local/lib/python3.8/site-packages/kserve/model.py", line 182, in _http_predict                                                                                           
kserve-container         response = await self._http_client.fetch(                                                                                                                                         
kserve-container     tornado.httpclient.HTTPClientError: HTTP 500: Internal Server Error                                                                                                                   
kserve-container [E 230303 08:31:33 web:2271] 500 POST /v1/models/sklearn-driver-transformer:predict (127.0.0.1) 45.16ms                                                                                   
queue-proxy {"severity":"INFO","timestamp":"2023-03-03T08:30:39.209761301Z","logger":"queueproxy","caller":"sharedmain/main.go:270","message":"Starting queue-proxy","commit":"d79b8bb-dirty","knative.dev 

-----------------
Container sklearn-driver-transformer-predictor-default-00001- logs

 queue-proxy {"severity":"INFO","timestamp":"2023-03-03T08:30:32.812574791Z","logger":"queueproxy","caller":"sharedmain/main.go:270","message":"Starting queue-proxy","commit":"d79b8bb-dirty","knative.dev 
 kserve-container 2023-03-03 08:30:34.705 1 root INFO [download():63] Copying contents of /mnt/models to local                                                                                              
 kserve-container /usr/local/lib/python3.9/site-packages/sklearn/base.py:324: UserWarning: Trying to unpickle estimator LinearRegression from version 1.0.2 when using version 1.0.1. This might lead to br 
 kserve-container https://scikit-learn.org/stable/modules/model_persistence.html#security-maintainability-limitations                                                                                       
 kserve-container   warnings.warn(                                                                                                                                                                          
 kserve-container 2023-03-03 08:30:35.094 1 root INFO [register_model():159] Registering model: sklearn-driver-transformer                                                                                  
 kserve-container 2023-03-03 08:30:35.095 1 root INFO [start():126] Setting max asyncio worker threads as 20                                                                                                
 kserve-container 2023-03-03 08:30:35.095 1 root INFO [serve():136] starting uvicorn with 1 workers                                                                                                         
 kserve-container 2023-03-03 08:30:35.126 1 root INFO [start():62] Starting gRPC server on :8081                                                                                                            
 kserve-container 2023-03-03 08:30:35.132 29 uvicorn.error INFO [serve():84] Started server process [29]                                                                                                    
 kserve-container 2023-03-03 08:30:35.133 29 uvicorn.error INFO [startup():45] Waiting for application startup.                                                                                             
 kserve-container 2023-03-03 08:30:35 DEBUG [timing_asgi.middleware:40] ASGI scope of type lifespan is not supported yet                                                                                    
 kserve-container 2023-03-03 08:30:35.134 29 uvicorn.error INFO [startup():59] Application startup complete.                                                                                                
 kserve-container /usr/local/lib/python3.9/site-packages/sklearn/base.py:445: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names                          
 kserve-container   warnings.warn(                                                                                                                                                                          
 kserve-container /usr/local/lib/python3.9/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated  
 kserve-container   X = check_array(X, **check_params)                                                                                                                                                      
 kserve-container 2023-03-03 08:31:33.614 29 root INFO [timing():49] kserve.io.kserve.protocol.rest.v1_endpoints.predict 0.0028722286224365234, ['http_status:500', 'http_method:POST', 'time:wall']        
 kserve-container 2023-03-03 08:31:33.614 29 root INFO [timing():49] kserve.io.kserve.protocol.rest.v1_endpoints.predict 0.002287000000000039, ['http_status:500', 'http_method:POST', 'time:cpu']          
 storage-initializer 2023-03-03 08:30:21.849 1 root INFO [<module>():13] Initializing, args: src_uri [gs://kfserving-examples/models/feast/driver] dest_path[ [/mnt/models]                                 
 storage-initializer 2023-03-03 08:30:21.849 1 root INFO [download():63] Copying contents of gs://kfserving-examples/models/feast/driver to local                                                           
 storage-initializer 2023-03-03 08:30:24.855 1 google.auth.compute_engine._metadata WARNING [ping():99] Compute Engine Metadata server unavailable on attempt 1 of 3. Reason: timed out                     
 storage-initializer 2023-03-03 08:30:27.858 1 google.auth.compute_engine._metadata WARNING [ping():99] Compute Engine Metadata server unavailable on attempt 2 of 3. Reason: timed out                     
 storage-initializer 2023-03-03 08:30:30.862 1 google.auth.compute_engine._metadata WARNING [ping():99] Compute Engine Metadata server unavailable on attempt 3 of 3. Reason: timed out                     
 storage-initializer 2023-03-03 08:30:30.862 1 google.auth._default WARNING [_get_gce_credentials():301] Authentication failed using Compute Engine authentication due to unavailable metadata server.      
 storage-initializer 2023-03-03 08:30:31.364 1 root INFO [_download_gcs():232] Downloading: /mnt/models/model.joblib                                                                                        
 storage-initializer 2023-03-03 08:30:31.445 1 root INFO [download():103] Successfully copied gs://kfserving-examples/models/feast/driver to /mnt/models                                                    
 Stream closed EOF for default/sklearn-driver-transformer-predictor-default-00001-deploymvw8xh (storage-initializer)                                                                                        
                                                                                                                                                                                                            
                                                                                                                                       
**What did you expect to happen:**


**What's the InferenceService yaml:**
apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: "sklearn-driver-transformer"
spec:
  transformer:
    containers:
    - image: padrian2s/driver-transformer:latest
      name: driver-container
      command:
      - "python"
      - "-m"
      - "driver_transformer"
      args:
      - --feast_serving_url
      - 172.18.255.200:6566
      - --entity_ids
      - driver_id
      - --feature_refs
      - driver_hourly_stats:acc_rate
      - driver_hourly_stats:avg_daily_trips
      - driver_hourly_stats:conv_rate
  predictor:
    sklearn:
      storageUri: "gs://kfserving-examples/models/feast/driver"
          

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Pip freeze on the sklearn-driver-transformer-transformer-default-00001 container


adal==1.2.7
aiohttp==3.8.4
aiohttp-cors==0.7.0
aioredis==1.3.1
aiorwlock==1.3.0
aiosignal==1.3.1
alog==0.9.13
anyio==3.6.2
async-timeout==4.0.2
attrs==22.2.0
avro==1.11.1
azure-core==1.26.3
azure-identity==1.12.0
azure-storage-blob==12.9.0
azure-storage-file-share==12.7.0
blessed==1.20.0
boto3==1.21.0
botocore==1.24.46
cachetools==5.3.0
certifi==2022.12.7
cffi==1.15.1
charset-normalizer==3.0.1
click==8.0.4
cloudevents==1.9.0
colorful==0.5.5
cryptography==39.0.2
deprecation==2.1.0
distlib==0.3.6
 Editable install with no version control (driver-transformer==0.2.0)
-e /
fastapi==0.88.0
filelock==3.9.0
frozenlist==1.3.3
google-api-core==2.11.0
google-auth==2.16.2
google-cloud-core==2.3.2
google-cloud-storage==2.7.0
google-crc32c==1.5.0
google-resumable-media==2.4.1
googleapis-common-protos==1.58.0
gpustat==1.0.0
grpcio==1.51.3
h11==0.14.0
hiredis==2.2.2
httpcore==0.16.3
httpx==0.23.3
idna==3.4
importlib-resources==5.12.0
isodate==0.6.1
jmespath==0.10.0
jsonschema==4.17.3
kserve==0.9.0
kubernetes==26.1.0
minio==7.1.3
msal==1.21.0
msal-extensions==1.0.0
msgpack==1.0.4
msrest==0.7.1
multidict==6.0.4
numpy==1.21.6
nvidia-ml-py==11.495.46
oauthlib==3.2.2
opencensus==0.11.1
opencensus-context==0.1.3
orjson==3.8.7
packaging==23.0
pkgutil-resolve-name==1.3.10
platformdirs==3.0.0
portalocker==2.7.0
prometheus-client==0.16.0
protobuf==3.19.6
psutil==5.9.4
py-spy==0.3.14
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
pydantic==1.10.5
PyJWT==2.6.0
pyrsistent==0.19.3
python-dateutil==2.8.2
python-rapidjson==1.9
PyYAML==6.0
ray==1.10.0
redis==4.5.1
requests==2.28.2
requests-oauthlib==1.3.1
rfc3986==1.5.0
rsa==4.9
s3transfer==0.5.2
six==1.16.0
smart-open==6.3.0
sniffio==1.3.0
starlette==0.22.0
table-logger==0.3.6
timing-asgi==0.3.0
tornado==6.2
tritonclient==2.18.0
typing-extensions==4.5.0
urllib3==1.26.14
uvicorn==0.20.0
virtualenv==20.20.0
wcwidth==0.2.6
websocket-client==1.5.1
yarl==1.8.2
zipp==3.15.0


**Environment:**

- Istio Version:
- Knative Version:
- KServe Version: 0.10  (kserve/release-0.10/hack/quick_install.sh)
- Kubeflow version: NA
- Cloud Environment:[kind]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2729 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:
NONE
```release-note

```

### CODE REVIEW

File: docs/samples/v1beta1/transformer/feast/redis.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+metadata:
+  name: redis-service
+spec:
+  type: LoadBalancer
+  selector:
+    app: redis-server
+  ports:
+    - protocol: TCP
+      port: 6379
+      targetPort: 6379
```
Comment (yuzisun) [inline docs/samples/v1beta1/transformer/feast/redis.yaml]: Nit: newline

File: docs/samples/v1beta1/transformer/feast/feature_store_initializer.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.9-slim
+RUN pip install --upgrade pip && pip install "feast[gcp]~=0.30.0" "feast[redis]~=0.30.0" "feast[aws]~=0.30.0"
+WORKDIR feature_store_initializer
+COPY feature_store_initializer_entrypoint.sh feature_store_initializer_entrypoint.sh
+RUN chmod +x feature_store_initializer_entrypoint.sh
+ENTRYPOINT ["./feature_store_initializer_entrypoint.sh"]
```
Comment (yuzisun) [inline docs/samples/v1beta1/transformer/feast/feature_store_initializer.Dockerfile]: Nit: newline

File: docs/samples/v1beta1/transformer/feast/feast_server.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+metadata:
+  name: feature-server-service
+spec:
+  type: LoadBalancer
+  selector:
+    app: feature-server
+  ports:
+    - protocol: TCP
+      port: 6566
+      targetPort: 6566
```
Comment (yuzisun) [inline docs/samples/v1beta1/transformer/feast/feast_server.yaml]: Nit: newline

File: docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer/driver_transformer.py [hunk 4]
```diff
@@ -101,20 +100,26 @@ def buildPredictRequest(self, inputs, features) -> Dict:
...
+        for i in range(len(features["results"][entity_ids_index]["values"])):
+            single_entity_data = [features["results"][acc_rate_index]["values"][i],
+                                  features["results"][avg_daily_trips_index]["values"][i],
+                                  features["results"][conv_rate_index]["values"][i],
+                                  features["results"][entity_ids_index]["values"][i]]
+            request_data.append(single_entity_data)
 
         # The default protocol is v1
         result = {'instances': request_data}
         if self.protocol == "v2":
```
Comment (yuzisun) [inline docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer/driver_transformer.py:119]: @sivanantha321 Can you help update this code with the latest 0.10 changes that you no longer need to handle rest/gRPC separately, if you return the unified type `InferRequest` then kserve can translate accordingly.

see example here https://github.com/kserve/kserve/blob/master/python/custom_transformer/model.py#L74

Comment (sivanantha321) [reply docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer/driver_transformer.py:119]: sure

**Comment by yuzisun** [comment]
Great update for the feast example! thanks @sivanantha321 !

/lgtm
/approve

**Pull request merged**